File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingGameChannel.java
package org.noxet.noxetserver.messaging.channels;

import org.bukkit.command.CommandSender;
import org.noxet.noxetserver.minigames.MiniGameController;

import java.util.HashSet;
import java.util.Set;

public class MessagingGameChannel extends MessagingChannel {
    private final MiniGameController miniGameController;

    public MessagingGameChannel(MiniGameController miniGameController) {
        this.miniGameController = miniGameController;
    }

    @Override
    public Set<CommandSender> getRecipients() {
        return new HashSet<>(miniGameController.getPlayersAndSpectators());
    }
}

File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingGameChannel.scala
package org.noxet.noxetserver.messaging.channels

import org.bukkit.command.CommandSender
import org.noxet.noxetserver.minigames.MiniGameController

import scala.collection.Set
import scala.jdk.CollectionConverters.*

class MessagingGameChannel(miniGameController: MiniGameController) extends MessagingChannel:
    override def getRecipients: Set[CommandSender] = miniGameController.getPlayersAndSpectators